### ISSUE TITLE
switchNetwork Is not working

### ISSUE THREAD

**Issue by Nilbristi** [Web3Auth/web3auth-wagmi-connector#47]
The **switchNetwork** method is not working when I am using it in **wagmi** **Web3AuthConnector**. But the same method is working fine with **wagmi** **MetaMaskConnector**

I am getting the below error:

```
{
    "code": -32603,
    "message": "t.match is not a function",
    "data": {
        "originalError": {}
    }
}
```

**Comment by raintreasure** [comment]
same error

**Comment by yashovardhan** [comment]
which version of `wagmi` are you using?

**Comment by Nilbristi** [comment]
@yashovardhan 
> 

{
  "name": "XXXXX",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build-dev": "ng build --configuration development",
    "build-prod": "ng build --configuration production",
    "watch": "ng build --watch --configuration development",
    "test": "ng test"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^14.2.12",
    "@angular/cdk": "^13.2.5",
    "@angular/common": "^14.2.12",
    "@angular/compiler": "^14.2.12",
    "@angular/core": "^14.2.12",
    "@angular/forms": "^14.2.12",
    "@angular/material": "^13.2.5",
    "@angular/platform-browser": "^14.2.12",
    "@angular/platform-browser-dynamic": "^14.2.12",
    "@angular/router": "^14.2.12",
    "@ngx-translate/core": "^14.0.0",
    "@ngx-translate/http-loader": "^7.0.0",
    "@wagmi/core": "^0.8.14",
    "@web3auth/base": "^4.6.0",
    "@web3auth/core": "^4.6.0",
    "@web3auth/modal": "^4.6.2",
    "@web3auth/openlogin-adapter": "^4.6.0",
    "@web3auth/torus-wallet-connector-plugin": "^4.6.0",
    "@web3auth/web3auth-wagmi-connector": "^2.2.0",
    "@wert-io/widget-initializer": "^2.0.3",
    "@wert-io/widget-sc-signer": "^1.0.0",
    "angular-svg-icon": "^13.0.0",
    "bootstrap": "^5.2.2",
    "browser": "^0.2.6",
    "ethers": "^5.7.2",
    "ngx-bootstrap": "^9.0.0",
    "ngx-hotjar": "^11.0.0",
    "ngx-owl-carousel-o": "^14.0.1",
    "rxjs": "~7.5.0",
    "sweetalert2": "^11.4.4",
    "tslib": "^2.3.0",
    "uuid": "^9.0.0",
    "web3": "^1.8.1",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^14.2.10",
    "@angular/cli": "^14.2.10",
    "@angular/compiler-cli": "^14.2.12",
    "@types/jasmine": "~3.10.0",
    "@types/node": "^12.11.1",
    "@types/uuid": "^9.0.0",
    "assert": "^2.0.0",
    "buffer": "^6.0.3",
    "crypto-browserify": "^3.12.0",
    "https-browserify": "^1.0.0",
    "jasmine-core": "~4.0.0",
    "karma": "~6.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.1.0",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "~1.7.0",
    "os-browserify": "^0.3.0",
    "process": "^0.11.10",
    "stream-browserify": "^3.0.0",
    "stream-http": "^3.2.0",
    "typescript": "~4.8.4",
    "url": "^0.11.0"
  },
  "peerDependencies": {
    "@ethereumjs/tx": "4.0.2"
  }
}

**Comment by chaitanyapotti** [comment]
Fixed in 3.0.0

**Comment by Nilbristi** [comment]
@chaitanyapotti I hope it will work fine with **Angular 14.2.12**. 
Thank you bro...
